doc/guides/platformio/index.rst

```diff
@@ -0,0 +1,61 @@
+.. _platformio:
+
+Using with PlatformIO
+#####################
+
+.. image:: img/platformio-logo.png
+  :target: https://platformio.org/?utm_source=docs.zephyrproject.org
+
+.. contents::
+    :local:
+
+What is PlatformIO?
+*******************
+
+`PlatformIO <https://platformio.org/?utm_source=docs.zephyrproject.org>`__
+is a cross-platform, cross-architecture, multiple framework tool for embedded
```

Comment:
Thanks for improving this considerably. I still think this reads a bit too much like an advertisement for PlatformIO, and we'd like to stay IDE and framework-neutral in Zephyr upstream.
Maybe @mbolivar-nordic can take a shot at shortening this a bit?

Reply:
Agreed, this is a lot better, thanks.  I will leave a suggestion.

Reply:
How about with this?
```
PlatformIO is a cross-platform and multiple architecture tool for embedded systems engineers and software developers who write applications for embedded
products. It is fully compatible with the Zephyr project.
```